### PR TITLE
Remove tekton resource validation

### PR DESCRIPTION
**Pull request by GijsvanDulmen**
# Changes

Makes it possible to create other resources in the trigger template besides Tekton resources. Solves https://github.com/tektoncd/triggers/issues/1696

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Removes the limit of only creating Tekton resources from within your TriggerTemplate. This will allow for other K8s resources to be created as well like for example K8s Jobs.
```

### CODE REVIEW

**Comment by dibyom** [comment]
looks like the builds tests are failing due to `goimports`:

```
pkg/apis/triggers/v1alpha1/trigger_template_validation.go:21: File is not `goimports`-ed (goimports)
	"fmt"
pkg/apis/triggers/v1beta1/trigger_template_validation.go:21: File is not `goimports`-ed (goimports)
	"fmt"
```

**Comment by GijsvanDulmen** [comment]
```release-note
Removes the limit of only creating Tekton resources from within your TriggerTemplate. This will allow for other K8s resources to be created as well like for example K8s Jobs.
```

File: docs/triggertemplates.md [hunk 1]
```diff
@@ -78,8 +78,7 @@ Keep the following in mind:
   * `tekton.dev/triggers-eventid`:`<EventID>` to track resources created by a specific event.
 
 * To support arbitrary resource types, Tekton resolves resource templates internally as byte blobs. Because of this, Tekton only validates these
-  resources when processing an event rather than at the creation of the `TriggerTemplate`. Thus, you can **only** specify Tekton resources in a
-  `TriggerTemplate`.
+  resources when processing an event rather than at the creation of the `TriggerTemplate`.
```
Comment (dibyom) [inline docs/triggertemplates.md:81]: I think its worth noting here that the default EL [cluster role](https://github.com/tektoncd/triggers/blob/main/config/200-clusterrole.yaml#L86-L108)  only allows creating Tekton resources by default. 
Users will need to add additional roles as necessary to the EventListener's SA in order to be able to create non Tekton resources.

**Review by dibyom — COMMENTED** [review]
A few small things, but overall looks good! Thanks @GijsvanDulmen 

cc @savitaashture @khrm

**Comment by GijsvanDulmen** [comment]
@dibyom Thanks for the review! Will process those asap :-)

File: examples/v1beta1/create-custom/create-custom-interceptor.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+apiVersion: triggers.tekton.dev/v1beta1
+kind: EventListener
+metadata:
+  name: create-custom-listener
```
[the remark this answers is not in the document]
Comment (GijsvanDulmen) [reply examples/v1beta1/create-custom/create-custom-interceptor.yaml]: Changed to create-configmap.

File: pkg/sink/sink.go [hunk 3]
```diff
@@ -61,6 +61,7 @@ type Sink struct {
 	KubeClientSet          kubernetes.Interface
 	TriggersClient         triggersclientset.Interface
 	DiscoveryClient        discoveryclient.ServerResourcesInterface
+	DynamicClientSet       dynamic.Interface
```
[the remark this answers is not in the document]
Comment (GijsvanDulmen) [reply pkg/sink/sink.go]: Nope, it's now gone.

**Review by dibyom — APPROVED** [review]
one tiny nit, otherwise LGTM! Thanks a lot @GijsvanDulmen

File: docs/triggertemplates.md [hunk 4]
```diff
@@ -78,8 +78,9 @@ Keep the following in mind:
   * `tekton.dev/triggers-eventid`:`<EventID>` to track resources created by a specific event.
 
 * To support arbitrary resource types, Tekton resolves resource templates internally as byte blobs. Because of this, Tekton only validates these
-  resources when processing an event rather than at the creation of the `TriggerTemplate`. Thus, you can **only** specify Tekton resources in a
-  `TriggerTemplate`.
+  resources when processing an event rather than at the creation of the `TriggerTemplate`. The default assigned cluster role to the `EventListener`
+  service account only allows creating Tekton resources. You will need to add the appropriate roles to the `EventListener` to be able to create other
```
[the remark this answers is not in the document]
Comment (GijsvanDulmen) [reply docs/triggertemplates.md]: Added!

**Comment by dibyom** [comment]
@savitaashture @khrm any objections to merging this?

**Comment by khrm** [comment]
Would take a look tomorrow.

File: pkg/apis/config/allowed_types.go [hunk 5]
```diff
@@ -1,43 +1,6 @@
...
-	return err
-}
-
 var (
-	AllowedPipelineTypes = map[string][]string{
-		"v1alpha1": {"pipelineruns", "taskruns", "pipelines", "clustertasks", "tasks", "conditions", "runs"},
-		"v1beta1":  {"pipelineruns", "taskruns", "pipelines", "clustertasks", "tasks", "customruns"},
-		"v1":       {"pipelineruns", "taskruns", "pipelines", "tasks"},
-	}
 	AllowedTriggersTypes = map[string][]string{
```
[the remark this answers is not in the document]
Comment (khrm) [reply pkg/apis/config/allowed_types.go:4]: OK. So this is used in WithClient function only.

File: pkg/client/dynamic/clientset/tekton/tekton.go [hunk 6]
```diff
@@ -28,16 +27,6 @@ import (
 // WithClient adds Tekton related clients to the Dynamic client.
 func WithClient(client dynamic.Interface) clientset.Option {
```
Comment (khrm) [inline pkg/client/dynamic/clientset/tekton/tekton.go:28]: I don't think we are using this function anywhere else apart from the places removed.

File: pkg/apis/config/allowed_types.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (khrm) [reply pkg/apis/config/allowed_types.go:4]: So please try removing this, if everything works fine, then we can remove this.

File: pkg/client/dynamic/clientset/tekton/tekton.go (hunk 6 above)
Comment (GijsvanDulmen) [reply pkg/client/dynamic/clientset/tekton/tekton.go]: @khrm seems to work!

**Review by khrm — COMMENTED** [review]
/lgtm

**Comment by khrm** [comment]
Everything looks fine. Let's merge this.

**Pull request merged**